protocol.json

```diff
@@ -0,0 +1,1216 @@
+{
+    "type": "object",
```

Comment:
Should this really be an object? Shouldn't the schema only define all structures?